### PR TITLE
Docker and circleci

### PR DESCRIPTION
**Pull request by djkapner**
Adopt after #37 

### Overview
Establishes a container and CI/CD of that container.
* This can free us from the confusing bamboo builds of this repo and the deployment of its conda.
* Pins python version in code, not in bamboo config.
* Can work on this without VPN connection for bamboo.
* Container more reproducible than conda: In writing a test case for #35 I found that I could locally demonstrate an edge case that fails with legacy code and is fixed by the new `preregister` step. But, that edge case passes on the bamboo build agent. 
* utilizing this new container in production will require changing the LIMS executable to a singularity call.

### contents
* Defines a docker container
* Defines a CircleCI build workflow
* Docker image is auto-built and pushed to dockerhub
* Docker image does not include conda, and has been demonstrated to run through singularity (see README).

### docker note
In the past, I have encountered some issues running a docker image through singularity when the image has a conda env inside of it. Conda attempts to write a tempfile internally to /opt/conda/<etc> which is allowed by docker, but, throws an OS permission error when tried through singularity. This implementation does not have conda inside the image.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM python:3.8.5
+
+ARG MYBRANCH
```
Comment (njmei) [inline Dockerfile:3]: This maybe should have a default value. If someone wants to build the docker image without invoking the CI.

https://docs.docker.com/engine/reference/builder/#default-values

Comment (djkapner) [reply Dockerfile]: done. set to master

**Review by njmei — APPROVED** [review]
LGTM
